treeherder/webapp/api/push.py

```diff
@@ -71,6 +70,20 @@ def list(self, request, project):
 
             pushes = pushes.filter(repository=repository)
 
+        search_param = filter_params.get("search")
+        if search_param:
+            filtered_commits = (
+                Commit.objects.annotate(
+                    search=SearchVector("revision", "author", "comments", config="english")
+                )
+                .filter(
+                    search=SearchQuery(search_param, config="english")
+                    # Get most recent results and limit result to 200
+                )
+                .order_by("-push__time")[:200]
```

Comment:
Are 200 commits sufficient? 150-200 commits on autoland per day is the base load with some pushes which update third party code or similar having >50 commits.

Reply:
Okay it was a suggestion to use 200 limit. What do you think is the appropriate limit to be use here?

Reply:
I suggested 200, but this was very arbitrary!
The goal is that I wanted to avoid a basically unlimited list in the `in` clause.

But note that this is the number of commits _after searching_.

What do you think would be appropriate @Archaeopteryx ?

Reply:
With the assumption a regression reached the release channel and gets noticed and investigated there after two weeks, it could have started 10 weeks ago.

10 weeks * 7 days / week * 160 commits (lower end used because less commits on the weekend) = 11200 commits.

You might know the characteristics when a regression gets investigated better and can adjust accordingly. We can also test with Treeherder's prototype instance but it has only 3 weeks of data because it needed a reset after migrations had diverged from production.

Reply:
I need to run it locally to see what SQL requests is generated from that,
but please note that we're not interested in the commits themselves, but in the distinct `push_id` values.
In that case the value will be much smaller IMO, and 200 is probably appropriate.

But maybe `.order_by("-push__time")[:200]` should be put after `.values_list` for this reason? Or maybe we need a `distinct` somewhere? I don't know django much so you might know better, otherwise I can look at this after lunch.

Reply:
1. Yes, add a [distinct](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#django.db.models.query.QuerySet.distinct) and limit the queryset at the end.
2. It is also reasonable to use the `Push.time` for condition, it's not expected the endpoint would be requested that often to cause an issue.

Reply:
Yes, filtering by `Push.time` prevents to miss commits in case many commits landed afterwards.

Reply:
I don't understand what you mean by "It is also reasonable to use the Push.time for condition", can you please elaborate?

Reply:
If the query is limited with `[:200]`, there are 200 matches (ordered by push time) for an unknown time frame. If there is a condition `push__time__gte=date.today() - timedelta(days=4 * 7 * 3)`, all pushes which landed in currently shipping products (ESR excluded) are searched.

Reply:
So retrieving pushes from the last ~3 months

Reply:
I think 200 (or more, the exact value could be tuned) matches with an unknown time frame sounds better in this case. For example it's fine (and even expected!) to retrieve older values if there's no match with the recent values.

Reply:
Ok, then let's keep as it is.